src/Illuminate/Database/Query/Builder.php

```diff
@@ -1469,6 +1470,58 @@ public function orWhereDate($column, $operator, $value = null)
         return $this->whereDate($column, $operator, $value, 'or');
     }
 
+    /**
+     * Add a where statement to the query after converting the date to app.timezone.
+     *
+     * @param  string  $column
+     * @param  string|Carbon  $operator
+     * @param  Carbon|null  $date
+     * @return $this
+     *
+     * @throws InvalidArgumentException
+     */
+    public function whereDateTime(string $column, string|Carbon $operator, Carbon|null $date = null)
+    {
+        [$date, $operator] = $this->prepareValueAndOperator(
+            $date, $operator, func_num_args() === 2
+        );
+
+        if (! ($date instanceof Carbon)) {
+            throw new InvalidArgumentException('Date must be a instance of Carbon');
+        }
+        if ($date->timezone->getName() != config('app.timezone')) {
```

Comment:
Wouldn't this break using just the Database component outside of Laravel application?

Reply:
yes, if an app does not have the config function or a default app.timezone this would not work.

are there limitations on what features are usable inside the query builder?